File: GO/ex1.go
package main

import "fmt"

func main() {
	var x, n, resultado int
	fmt.Print("Digite o valor de x: ")
	fmt.Scan(&x)
	fmt.Print("Digite o valor de n (não-negativo): ")
	fmt.Scan(&n)

	if n < 0 {
		fmt.Println("n não é compatível")
		return
	}

	resultado = 1
	for i := 0; i < n; i++ {
		resultado *= x
	}

	fmt.Printf("%d elevado a %d é igual a: %d\n", x, n, resultado)
}

File: Haskell Script/ex1.hs
main :: IO ()
main = do
    putStrLn "Digite o valor de x: "
    x <- readLn :: IO Int
    n <- solicitarPositivo "Digite o valor de n (não-negativo): "
    let resultado = x ^ n
    putStrLn $ show x ++ " elevado a " ++ show n ++ " é igual a: " ++ show resultado

solicitarPositivo :: String -> IO Int
solicitarPositivo prompt = do
    putStrLn prompt
    n <- readLn :: IO Int
    if n < 0
        then do
            putStrLn "Por favor, digite um número não-negativo."
            solicitarPositivo prompt
        else return n
